### PR TITLE
add setConnectData method

### PR DESCRIPTION
**Pull request by larry-dev**
add setConnectData method

### CODE REVIEW

File: centrifuge/src/main/java/io/github/centrifugal/centrifuge/Client.java [hunk 1]
```diff
@@ -44,6 +44,7 @@ Options getOpts() {
 
     private Options opts;
     private String token = "";
+    private com.google.protobuf.ByteString connectData;
```
Comment (FZambia) [inline centrifuge/src/main/java/io/github/centrifugal/centrifuge/Client.java:47]: I suppose it's better to use `byte[]` like we do [in RPC method](https://github.com/centrifugal/centrifuge-java/blob/master/centrifuge/src/main/java/io/github/centrifugal/centrifuge/Client.java#L786) for example?

Comment (FZambia) [reply centrifuge/src/main/java/io/github/centrifugal/centrifuge/Client.java:47]: ah sorry, I am wrong - you already use it right.

File: centrifuge/src/main/java/io/github/centrifugal/centrifuge/Client.java [hunk 2]
```diff
@@ -614,9 +625,12 @@ public void onDone(ReplyError error, Protocol.RefreshResult result) {
     }
 
     private void sendConnect() {
-        Protocol.ConnectRequest req = Protocol.ConnectRequest.newBuilder()
-                .setToken(this.token)
-                .build();
+        Protocol.ConnectRequest.Builder build=Protocol.ConnectRequest.newBuilder()
```
Comment (FZambia) [inline centrifuge/src/main/java/io/github/centrifugal/centrifuge/Client.java:628]: I suppose sth wrong with formatting here and below? I mean no spaces between tokens.

Comment (FZambia) [reply centrifuge/src/main/java/io/github/centrifugal/centrifuge/Client.java:628]: I just mean that you should use the same code formatting with spaces between `=` for example. And in other places.

File: centrifuge/src/main/java/io/github/centrifugal/centrifuge/Client.java [hunk 3]
```diff
@@ -21,10 +21,7 @@
 import com.google.gson.JsonObject;
 import com.google.protobuf.InvalidProtocolBufferException;
 
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
+import java.util.*;
```
Comment (FZambia) [inline centrifuge/src/main/java/io/github/centrifugal/centrifuge/Client.java:24]: Lets stay specific in what we are importing.

File: centrifuge/src/main/java/io/github/centrifugal/centrifuge/Client.java [hunk 4]
```diff
@@ -659,6 +673,10 @@ private void handleAsyncReply(Protocol.Reply reply) {
                     PublishEvent event = new PublishEvent();
                     event.setData(pub.getData().toByteArray());
                     sub.getListener().onPublish(sub, event);
+                }else if(this.serverSide.containsKey(channel)){
+                    MessageEvent event = new MessageEvent();
```
Comment (FZambia) [inline centrifuge/src/main/java/io/github/centrifugal/centrifuge/Client.java:677]: Server side subs implementation is wrong -  we need new handlers on top level for server side publications. I suggest only add connect data in this pull request and continue work on server side subs in another pr. I can explain a bit more how this should work and why in Discord. What do you think?

File: centrifuge/src/main/java/io/github/centrifugal/centrifuge/Client.java (hunk 3 above)
Comment (larry-dev) [reply centrifuge/src/main/java/io/github/centrifugal/centrifuge/Client.java:24]: ok

File: centrifuge/src/main/java/io/github/centrifugal/centrifuge/Client.java (hunk 4 above)
Comment (larry-dev) [reply centrifuge/src/main/java/io/github/centrifugal/centrifuge/Client.java]: @FZambia i have removed server side subs implementation.  and formatter set connect data code.

File: centrifuge/src/main/java/io/github/centrifugal/centrifuge/Client.java [hunk 5]
```diff
@@ -84,9 +86,22 @@ public void setToken(String token) {
...
 
+    /**
+     * Set connection data This is a data you have to receive from your application backend.
+     *
+     * @param data
+     */
+    public void setConnectData(byte[] data) {
+        this.executor.submit(() -> {
+            Client.this.connectData = com.google.protobuf.ByteString.copyFrom(data);
+            ;
```
Comment (FZambia) [inline centrifuge/src/main/java/io/github/centrifugal/centrifuge/Client.java:97]: unnecessary semicolon?

File: centrifuge/src/main/java/io/github/centrifugal/centrifuge/Client.java [hunk 6]
```diff
@@ -154,6 +169,7 @@ public void onClosing(WebSocket webSocket, int code, String reason) {
             @Override
             public void onClosed(WebSocket webSocket, int code, String reason) {
                 super.onClosed(webSocket, code, reason);
+                //Clear serverSide sub
```
Comment (FZambia) [inline centrifuge/src/main/java/io/github/centrifugal/centrifuge/Client.java:172]: unnecessary comment left here

Comment (larry-dev) [reply centrifuge/src/main/java/io/github/centrifugal/centrifuge/Client.java]: sorry,  and i remove unnecessary comment.

**Comment by FZambia** [comment]
@larry-dev one more unresolved review comment about semicolon - please take a look

**Comment by larry-dev** [comment]
sorry,i missed the message.and i resolved this question.

**Comment by FZambia** [comment]
Thanks!

**Pull request merged**